homeassistant/components/asuswrt/config_flow.py

```diff
@@ -143,18 +143,17 @@ def _show_setup_form(
             errors=errors or {},
         )
 
-    @staticmethod
     async def _async_check_connection(
-        user_input: dict[str, Any]
+        self, user_input: dict[str, Any]
     ) -> tuple[str, str | None]:
         """Attempt to connect the AsusWrt router."""
 
         host: str = user_input[CONF_HOST]
-        api = get_api(user_input)
+        api = AsusWrtBridge.get_bridge(self.hass, user_input)
         try:
-            await api.connection.async_connect()
+            await api.async_connect()
 
-        except OSError:
+        except ConfigEntryNotReady:
```

Comment:
I think we shouldn't catch `ConfigEntryNotReady` in the config flow. Maybe we should catch `OSError` here, not in  `async_connect()`?

Reply:
I raised `ConfigEntryNotReady` at bridge level in case of connection error to have generalized exception that will allow common code in the config flow (and not related to specific library exception). If is better to not catch `ConfigEntryNotReady` we could raise a new dedicated exception in bridge module and catch that one. Let me know.

Reply:
IMO the best solution is catching `OSError` in the backend library and raise library specific error. `ConfigEntryNotReady` should be raised only in `async_setup_entry` https://developers.home-assistant.io/docs/integration_setup_failures?_highlight=configentrynotready#integrations-using-async_setup_entry

Reply:
The problem is that change the old backend library (`aioasuswrt`) is not so immediate and in any case when we will integrate a second backend library to add http protocol we will have 2 different exception. The idea is to have a unique exception that is not related to the library for all the modules hat are not the bridge. But of course we could use a different exception that is not `ConfigEntryNotReady`.

Reply:
Yeah I understand that. That's why I suggested catching `OSError` here not in `async_connect()` in the first comment.

Reply:
Ok, I used `OSError` as exception in both config flow and Router initialization